src/sr.ml

```diff
@@ -41,12 +41,9 @@ let subst_from_constrs : constrs -> tvar array * term array = fun cs ->
         let (hb,argsb) = get_args b in
         match (unfold ha, unfold hb) with
         | (Symb(Sym(sa)), Symb(Sym(sb))) when sa == sb ->
-            let cs =
-              try List.combine argsa argsb @ cs with Invalid_argument _ -> cs
-            in
+            let cs = try List.combine argsa argsb @ cs with _ -> cs in
```

Comment:
Wouldn't it be better to write:
   ```with Invalid_argument _ -> assert false (* cannot happen *)```
?

Reply:
What if sa and sb are not applied to the same number of arguments (even if sa==sb)? This could happen.

Reply:
Line 57: use List.split?

Reply:
Line 53: put a more explicit warning?

Reply:
As you mentioned, this can happen, so we cannot use an `assert false` here. If the number of arguments is different, then the constraint is no good for building a substitution, and there is nothing we can do with it so it is ignored. In fact, we could fail because this means that the constraints cannot be solved.

An any case, this is not really a bug, because it just means that we accept to type-check rules that we will never be able to apply.

Reply:
Line 72: rename t into l and u into r?

Reply:
About assert false: Then, it would be good to give more explanations about what this function does.